src/main/riddl/Events/eventContext.riddl

```diff
@@ -0,0 +1,197 @@
+                on command ChangeEventInfo{
+                    then "set entity eventInfo to message info"
+                    and "set entity lastModifiedBy to Event.cancellingMember"
+                    and "set entity lastModifiedOn to today()"
+                }
+                on command CancelEvent{
+                    then "morph entity Event to state EventStatus.Cancelled"
+                    and "set entity lastModifiedBy to Event.cancellingMember"
+                    and "set entity lastModifiedOn to today()"
+                }
+                on command RescheduleEvent {
+                  then "set entity evenInfo.expectedStart to Event.start"
+                  and "set entity evenInfo.expectedEnd to Event.end"
+                  and "set entity lastModifiedBy to Event.reschedulingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command DelayEvent{
+                  then "morph Event entity to EventStatus.Delayed"
+                  and "set entity eventInfo.expectedStart to eventInfo.expectedStart + Event.duration"
+                  and "set entity eventInfo.expectedEnd to eventInfo.expectedEnd + Event.duration"
+                  and "set entity lastModifiedBy to Event.delayingMember"
+                  and "set entity lastModifiedOn to today()"
+                  // what about Event.reason? Do we want to store it somewhere?
+                }
+                on command StartEvent{
+                  then "morph Event entity to EventStatus.InProgress"
+                  and "set entity lastModifiedBy to Event.startingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command EndEvent{
+									then error "End has not started. Cannot end an event that has not started."
+                }
+                on command AddLiveUpdate{???}
+            }
+        }
+        state InProgress is {
+            fields{
+                eventId: EventId,
+                eventInfo: EventInfo,
+                eventMetaInfo: EventMetaInfo
+            }
+            handler InProgressEventHandler is {
+                on command ScheduleEvent{ ??? // is this allowed or does an event have to be delayed first?
+                }
+                on command ChangeEventInfo{ ??? // can info be changed while event is in progress? Would this be more likely to confuse or help users?
+                }
+                on command CancelEvent{
+									then "morph Event entity to EventStatus.Cancelled"
+                  and "set entity lastModifiedBy to Event.cancellingMember"
+                  and "set entity lastModifiedOn to today()"
+                }
+                on command RescheduleEvent{ ??? // is this allowed or does the event have to be delayed first?
+                }
+                on command DelayEvent{ ??? // is this allowed or return error?
```

Comment:
Yes. You can delay an InProgress event. I think InProgress may be the only state that you can delay an event in. In other states you would just reschedule.